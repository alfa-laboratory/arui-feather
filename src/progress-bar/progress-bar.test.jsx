/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { mount } from 'enzyme';
import ProgressBar from './progress-bar';

describe('progress-bar', () => {
    it('should render without problems', () => {
        let progressBar = mount(<ProgressBar />);

        expect(progressBar).toMatchSnapshot();
    });

    it('should display current value', () => {
        let progressBar = mount(<ProgressBar percent={ 50 } />);
        let currentValueWrapper = progressBar.find('.progress-bar__current-value');

        expect(currentValueWrapper.getDOMNode().getAttribute('style')).toEqual('width: 50%;');
    });
});
