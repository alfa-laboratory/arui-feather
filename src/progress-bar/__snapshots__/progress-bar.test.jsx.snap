// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress-bar should render without problems 1`] = `
<ProgressBar
  percent={0}
  size="m"
>
  <div
    className="progress-bar progress-bar_size_m progress-bar_theme_alfa-on-white"
  >
    <div
      className="progress-bar__current-value"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</ProgressBar>
`;
